### PR TITLE
Separate version hash from version number in editor and project manager

### PR DESCRIPTION
**Pull request by Xrayez**
When copy-pasting the version from About dialog to bug reports at GitHub, this makes the version hash linkable to commits at GitHub (users might not be aware of this feature):

image

v4.0.dev.Xrayez a2b6336b1.

### CODE REVIEW

**Comment by Calinou** [comment]
This needs to be done in 3 places:

- Editor About dialog (`editor/editor_about.cpp`, what you did here).
- Project Manager (`editor/project_manager.cpp`).
- Editor bottom panel (`editor/editor_node.cpp`).

**Comment by Xrayez** [comment]
Done, I didn't realize it was possible to copy version directly from within the bottom panel as well, and I've never actually paid attention to version number in the project manager. 😃

**Comment by Xrayez** [comment]
Here's how it looks with `()`:

image

image

Haven't pushed the changes yet.

**Comment by Xrayez** [comment]
Square brackets `[]`:

image

image

File: editor/editor_about.cpp [hunk 1]
```diff
@@ -141,7 +141,7 @@ EditorAbout::EditorAbout() {
 	version_btn = memnew(LinkButton);
 	String hash = String(VERSION_HASH);
 	if (hash.length() != 0) {
-		hash = "." + hash.left(9);
+		hash = " " + vformat("[%s]", hash.left(9));
```
Comment (Xrayez) [inline editor/editor_about.cpp:144]: Pushed square brackets version, looks good to me. I could embed the space directly in `vformat`, but then I think it's better to be explicit here, so it doesn't seem like a typo.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.4.

**Comment by akien-mga** [comment]
Cherry-picked for 3.3.3.

**Pull request merged**